File: Java/Implementation/ApplesAndOranges/src/ApplesAndOranges.java
/*
 * Hackerrank Problem - Apple and Orange (Implementation Category)
 * 
 * @author ketanmalik
 * May 14 2020
 * 
 * */

public class ApplesAndOranges {
	public static void main(String args[]) {
		countApplesAndOranges(7, 11, 5, 15, new int[] { -2, 2, 1 }, new int[] { 5, -6 });
	}

	static void countApplesAndOranges(int s, int t, int a, int b, int[] apples, int[] oranges) {
		int applesCount = 0;
		int orangesCount = 0;
		int dist = 0;
		for (int i = 0; i < apples.length; i++) {
			dist = apples[i] + a;
			if (dist >= s && dist <= t)
				applesCount++;
			dist = 0;
		}
		for (int i = 0; i < oranges.length; i++) {
			dist = oranges[i] + b;
			if (dist >= s && dist <= t)
				orangesCount++;
			dist = 0;
		}
		System.out.println(applesCount);
		System.out.println(orangesCount);
	}
}

File: JS/Implementation/ApplesAndOranges.js
/*
 * Hackerrank Problem - Apple and Orange (Implementation Category)
 *
 * @author ketanmalik
 * May 14 2020
 *
 * */

var applesAndOranges = function (s, t, a, b, apples, oranges) {
  var applesCount = 0;
  var orangesCount = 0;
  var dist = 0;
  for (var x in apples) {
    dist = apples[x] + a;
    dist >= s && dist <= t ? applesCount++ : "";
  }
  for (var x in oranges) {
    dist = oranges[x] + b;
    dist >= s && dist <= t ? orangesCount++ : "";
  }
  console.log(applesCount);
  console.log(orangesCount);
};

applesAndOranges(7, 11, 5, 15, [-2, 2, 1], [5, -6]);
